extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -149,16 +150,41 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
       return;
     }
 
-    this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
     this.onRtlStateUpdate_(this.storeService_.get(StateProperty.RTL_STATE));
-    this.mutateElement(() => {
-      createShadowRootWithStyle(
-        this.element,
-        this.shoppingTagEl_,
-        shoppingTagCSS
-      );
-      this.hasAppendedInnerShoppingTagEl_ = true;
-    });
+
+    this.measureMutateElement(
+      () => {
+        this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
+      },
+      () => {
+        createShadowRootWithStyle(
+          this.element,
+          this.shoppingTagEl_,
+          shoppingTagCSS
+        );
+        this.hasAppendedInnerShoppingTagEl_ = true;
+
+        const edgeOfScreentoTagDistance =
+          document.getElementsByTagName('amp-story-shopping-attachment')[0]
+            ./*OK*/ clientWidth *
+            0.5 -
+          this.element./*OK*/ offsetLeft -
+          this.element./*OK*/ offsetWidth * 0.5;
+
+        const shoppingTagInner = this.element.shadowRoot.querySelector(
+          '.amp-story-shopping-tag-inner'
+        );
+        if ( (edgeOfScreentoTagDistance < -160 && this.shoppingTagEl_.getAttribute('dir') !== 'rtl')
+        || (edgeOfScreentoTagDistance > 145 && this.shoppingTagEl_.getAttribute('dir') === 'rtl')
+        ) {
+          shoppingTagInner?.classList.add('amp-story-shopping-tag-inner-flipped'); 
+          shoppingTagInner?.classList.remove('.amp-story-shopping-tag-inner');
+        } else {
+          shoppingTagInner?.classList.add('amp-story-shopping-tag-inner-flipped');
+          shoppingTagInner?.classList.remove('amp-story-shopping-tag-inner-flipped');
```

Comment:
Instead of adding and removing, we could toggle an `amp-story-shopping-tag-inner-flipped` class that overrides the flex direction. That will simplify this logic.